In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')


In [ ]:
df = pd.read_csv('https://raw.githubusercontent.com/ArchanaInsights/Datasets/refs/heads/main/crowdfunding_campaign.csv')

#Data Preprocessing

In [ ]:
df.head()

CampaignID  GoalAmount  RaisedAmount  DurationDays  NumBackers Category  \
0  CID_00000       76159  73518.028955            15        1903     Film   
1  CID_00001       69394  60068.683288            49         469      Art   
2  CID_00002       18652  22958.153470            58         148     Film   
3  CID_00003       19487  24229.048390            39         780     Film   
4  CID_00004       26734  17984.630311            68        1622    Games   

  LaunchMonth    Country Currency  OwnerExperience VideoIncluded  \
0   September  Australia      GBP               11           Yes   
1     October  Australia      GBP                7           Yes   
2    February        USA      EUR               17            No   
3     October    Germany      AUD               10            No   
4   September  Australia      AUD               12           Yes   

   SocialMediaPresence  NumUpdates  IsSuccessful  
0                41758           3             0  
1                79112          48             0  
2                 3942           9             1  
3                22047          35             1  
4                72568          17             0

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 14 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   CampaignID           100000 non-null  object 
 1   GoalAmount           100000 non-null  int64  
 2   RaisedAmount         100000 non-null  float64
 3   DurationDays         100000 non-null  int64  
 4   NumBackers           100000 non-null  int64  
 5   Category             100000 non-null  object 
 6   LaunchMonth          100000 non-null  object 
 7   Country              100000 non-null  object 
 8   Currency             100000 non-null  object 
 9   OwnerExperience      100000 non-null  int64  
 10  VideoIncluded        100000 non-null  object 
 11  SocialMediaPresence  100000 non-null  int64  
 12  NumUpdates           100000 non-null  int64  
 13  IsSuccessful         100000 non-null  int64  
dtypes: float64(1), int64(7), object(6)
memory usage: 10.7+ MB


In [ ]:
df.describe().round()

CampaignID  GoalAmount  RaisedAmount  DurationDays  NumBackers  \
count    100000.0    100000.0      100000.0      100000.0    100000.0   
mean      50000.0     50538.0       50598.0          52.0      1003.0   
std       28868.0     28614.0       33255.0          22.0       574.0   
min           0.0      1000.0         529.0          15.0        10.0   
25%       25000.0     25725.0       23397.0          33.0       505.0   
50%       50000.0     50706.0       45981.0          52.0      1003.0   
75%       74999.0     75316.0       72790.0          71.0      1502.0   
max       99999.0     99999.0      149238.0          89.0      1999.0   

       Category  LaunchMonth   Country  Currency  OwnerExperience  \
count  100000.0     100000.0  100000.0  100000.0         100000.0   
mean        2.0          6.0       2.0       2.0              9.0   
std         1.0          3.0       1.0       1.0              6.0   
min         0.0          0.0       0.0       0.0              0.0   
25%         1.0          2.0       1.0       1.0              4.0   
50%         2.0          6.0       2.0       2.0              9.0   
75%         3.0          8.0       3.0       3.0             14.0   
max         4.0         11.0       4.0       4.0             19.0   

       VideoIncluded  SocialMediaPresence  NumUpdates  IsSuccessful  
count       100000.0             100000.0    100000.0      100000.0  
mean             0.0              50077.0        24.0           1.0  
std              0.0              28768.0        14.0           1.0  
min              0.0                100.0         0.0           0.0  
25%              0.0              25277.0        12.0           0.0  
50%              0.0              50024.0        25.0           1.0  
75%              1.0              74934.0        37.0           1.0  
max              1.0              99999.0        49.0           1.0

In [ ]:
df.isnull().sum()

CampaignID             0
GoalAmount             0
RaisedAmount           0
DurationDays           0
NumBackers             0
Category               0
LaunchMonth            0
Country                0
Currency               0
OwnerExperience        0
VideoIncluded          0
SocialMediaPresence    0
NumUpdates             0
IsSuccessful           0
dtype: int64

In [ ]:
df.dtypes

CampaignID              object
GoalAmount               int64
RaisedAmount           float64
DurationDays             int64
NumBackers               int64
Category                object
LaunchMonth             object
Country                 object
Currency                object
OwnerExperience          int64
VideoIncluded           object
SocialMediaPresence      int64
NumUpdates               int64
IsSuccessful             int64
dtype: object

In [ ]:
from sklearn.preprocessing import LabelEncoder
le = LabelEncoder()
le.fit(df['Category'])
df['Category'] = le.transform(df['Category'])
df.head().round()

CampaignID  GoalAmount  RaisedAmount  DurationDays  NumBackers  Category  \
0  CID_00000       76159       73518.0            15        1903         1   
1  CID_00001       69394       60069.0            49         469         0   
2  CID_00002       18652       22958.0            58         148         1   
3  CID_00003       19487       24229.0            39         780         1   
4  CID_00004       26734       17985.0            68        1622         2   

   LaunchMonth  Country Currency  OwnerExperience VideoIncluded  \
0           11        0      GBP               11           Yes   
1           10        0      GBP                7           Yes   
2            3        4      EUR               17            No   
3           10        2      AUD               10            No   
4           11        0      AUD               12           Yes   

   SocialMediaPresence  NumUpdates  IsSuccessful  
0                41758           3             0  
1                79112          48             0  
2                 3942           9             1  
3                22047          35             1  
4                72568          17             0

In [ ]:
le.fit(df['LaunchMonth'])
df['LaunchMonth'] = le.transform(df['LaunchMonth'])
df.head().round()

CampaignID  GoalAmount  RaisedAmount  DurationDays  NumBackers  Category  \
0  CID_00000       76159       73518.0            15        1903         1   
1  CID_00001       69394       60069.0            49         469         0   
2  CID_00002       18652       22958.0            58         148         1   
3  CID_00003       19487       24229.0            39         780         1   
4  CID_00004       26734       17985.0            68        1622         2   

   LaunchMonth  Country Currency  OwnerExperience VideoIncluded  \
0           11        0      GBP               11           Yes   
1           10        0      GBP                7           Yes   
2            3        4      EUR               17            No   
3           10        2      AUD               10            No   
4           11        0      AUD               12           Yes   

   SocialMediaPresence  NumUpdates  IsSuccessful  
0                41758           3             0  
1                79112          48             0  
2                 3942           9             1  
3                22047          35             1  
4                72568          17             0

In [ ]:
le.fit(df['Country'])
df['Country'] = le.transform(df['Country'])
df.head().round()

CampaignID  GoalAmount  RaisedAmount  DurationDays  NumBackers  Category  \
0  CID_00000       76159       73518.0            15        1903         1   
1  CID_00001       69394       60069.0            49         469         0   
2  CID_00002       18652       22958.0            58         148         1   
3  CID_00003       19487       24229.0            39         780         1   
4  CID_00004       26734       17985.0            68        1622         2   

   LaunchMonth  Country Currency  OwnerExperience VideoIncluded  \
0           11        0      GBP               11           Yes   
1           10        0      GBP                7           Yes   
2            3        4      EUR               17            No   
3           10        2      AUD               10            No   
4           11        0      AUD               12           Yes   

   SocialMediaPresence  NumUpdates  IsSuccessful  
0                41758           3             0  
1                79112          48             0  
2                 3942           9             1  
3                22047          35             1  
4                72568          17             0

In [ ]:
le.fit(df['Currency'])
df['Currency'] = le.transform(df['Currency'])
df.head().round()

CampaignID  GoalAmount  RaisedAmount  DurationDays  NumBackers  Category  \
0  CID_00000       76159       73518.0            15        1903         1   
1  CID_00001       69394       60069.0            49         469         0   
2  CID_00002       18652       22958.0            58         148         1   
3  CID_00003       19487       24229.0            39         780         1   
4  CID_00004       26734       17985.0            68        1622         2   

   LaunchMonth  Country  Currency  OwnerExperience VideoIncluded  \
0           11        0         3               11           Yes   
1           10        0         3                7           Yes   
2            3        4         2               17            No   
3           10        2         0               10            No   
4           11        0         0               12           Yes   

   SocialMediaPresence  NumUpdates  IsSuccessful  
0                41758           3             0  
1                79112          48             0  
2                 3942           9             1  
3                22047          35             1  
4                72568          17             0

In [ ]:
le.fit(df['VideoIncluded'])
df['VideoIncluded'] = le.transform(df['VideoIncluded'])
df.head().round()

CampaignID  GoalAmount  RaisedAmount  DurationDays  NumBackers  Category  \
0  CID_00000       76159       73518.0            15        1903         1   
1  CID_00001       69394       60069.0            49         469         0   
2  CID_00002       18652       22958.0            58         148         1   
3  CID_00003       19487       24229.0            39         780         1   
4  CID_00004       26734       17985.0            68        1622         2   

   LaunchMonth  Country  Currency  OwnerExperience  VideoIncluded  \
0           11        0         3               11              1   
1           10        0         3                7              1   
2            3        4         2               17              0   
3           10        2         0               10              0   
4           11        0         0               12              1   

   SocialMediaPresence  NumUpdates  IsSuccessful  
0                41758           3             0  
1                79112          48             0  
2                 3942           9             1  
3                22047          35             1  
4                72568          17             0

In [ ]:
le.fit(df['CampaignID'])
df['CampaignID'] = le.transform(df['CampaignID'])
df.head().round()

CampaignID  GoalAmount  RaisedAmount  DurationDays  NumBackers  Category  \
0           0       76159       73518.0            15        1903         1   
1           1       69394       60069.0            49         469         0   
2           2       18652       22958.0            58         148         1   
3           3       19487       24229.0            39         780         1   
4           4       26734       17985.0            68        1622         2   

   LaunchMonth  Country  Currency  OwnerExperience  VideoIncluded  \
0           11        0         3               11              1   
1           10        0         3                7              1   
2            3        4         2               17              0   
3           10        2         0               10              0   
4           11        0         0               12              1   

   SocialMediaPresence  NumUpdates  IsSuccessful  
0                41758           3             0  
1                79112          48             0  
2                 3942           9             1  
3                22047          35             1  
4                72568          17             0

#Data Splitting

In [ ]:
features = ['CampaignID','GoalAmount','RaisedAmount','DurationDays','NumBackers','Category','LaunchMonth','Country','Currency','OwnerExperience','VideoIncluded','SocialMediaPresence','NumUpdates']
target = 'IsSuccessful'

X = df[features]
Y = df[target]


In [ ]:
from sklearn.model_selection import train_test_split

X_train,X_test,Y_train,Y_test = train_test_split(X,Y,train_size= 0.7,test_size = 0.2,random_state= 42)

print(f'Dataset Shape: {df.shape}')

Dataset Shape: (100000, 14)


In [ ]:
X_train.shape

(70000, 13)

In [ ]:
X_test.shape

(20000, 13)

In [ ]:
Y_train.shape

(70000,)

In [ ]:
Y_test.shape

(20000,)

#Feature Scaling

In [ ]:
from sklearn.preprocessing import StandardScaler

scaler = StandardScaler()

scaler.fit(X_train)
scaler.fit(X_test)

X_train_scaled = scaler.transform(X_train)
X_test_scaled = scaler.transform(X_test)

print(X_train_scaled)
print(X_test_scaled)

[[ 0.87701416  0.99887961  1.35458408 ... -0.99084193  0.53346327
  -0.87230841]
 [-0.03308421  0.35257377  1.065371   ...  1.00924271 -1.29103639
   0.79930755]
 [-0.1713055  -0.47655872 -0.95014445 ...  1.00924271  0.10436392
  -1.42951373]
 ...
 [-1.17184197 -1.39654693 -1.1196399  ... -0.99084193  0.49845111
  -1.22056174]
 [ 0.63705866 -1.25846855 -1.16198816 ...  1.00924271  0.57035669
  -1.63846573]
 [ 0.82593915  1.33560686  1.05415421 ...  1.00924271 -0.17691791
  -0.73300708]]
[[ 0.89437411  0.37062642 -0.40762232 ...  1.00924271 -0.58065521
   0.38140356]
 [ 1.0490198  -1.3640661  -1.25370643 ... -0.99084193  1.18862626
   0.79930755]
 [-1.0411052  -1.30551696 -1.10512551 ... -0.99084193 -0.57640498
   1.63511553]
 ...
 [-0.41746346 -0.32774632  0.22443246 ...  1.00924271  1.55776946
   0.59035555]
 [ 0.12166543 -1.65161905 -1.47173654 ...  1.00924271 -1.07615575
  -1.08126041]
 [-0.25987247 -0.41727772  0.15209778 ... -0.99084193 -1.14384594
  -1.35986307]]


#Random Forest - Model Building and Evaluation

In [ ]:
from sklearn.ensemble import RandomForestClassifier

model = RandomForestClassifier(n_estimators=100, random_state=24)
model.fit(X_train, Y_train)

RandomForestClassifier(random_state=24)

In [ ]:
n_estimators_values = [50, 100]
max_depth_values = [5, 10]

for n_estimators in n_estimators_values:
    for max_depth in max_depth_values:
        rf = RandomForestClassifier(n_estimators=n_estimators, max_depth=max_depth, random_state=42)
        rf.fit(X_train, Y_train)

In [ ]:
Y_pred = model.predict(X_test)

In [ ]:
from sklearn.metrics import accuracy_score

model = RandomForestClassifier(n_estimators=100, random_state=24,max_depth=5)

print("Accuracy:")
print(accuracy_score(Y_test, Y_pred))

Accuracy:
0.9682


#Naive Bayes - Model Building and Evaluation

In [ ]:
from sklearn.naive_bayes import GaussianNB
nb_model=GaussianNB()
nb_model.fit(X_train_scaled,Y_train)

GaussianNB()

In [ ]:
Y_pred = nb_model.predict(X_test_scaled)

In [ ]:
from sklearn.metrics import accuracy_score
accuracy = accuracy_score(Y_test, Y_pred)
print("Accuracy:", accuracy_score(Y_test, Y_pred))

Accuracy: 0.6613


#Support Vector Machine (SVM) - Model Building and Evaluation

In [ ]:
from sklearn.svm import SVC
svm_classifier = SVC(kernel='linear', random_state=42)
svm_classifier.fit(X_train, Y_train)

SVC(kernel='linear', random_state=42)

In [ ]:
Y_pred = svm_classifier.predict(X_test_scaled)

In [ ]:
from sklearn.metrics import accuracy_score
accuracy = accuracy_score(Y_test, Y_pred)
print("Accuracy:", accuracy_score(Y_test, Y_pred))

Accuracy: 0.50235


#Comparison and Analysis:


  5.1.	Compare the accuracy scores of the three models.

 Random Forest model: 96%

 Naive Bayes model  : 66%

 SVM model          : 50%

  5.2. Discuss which model performed best and why,comparing to the models based on their accuracy Random Forest model is best.
  Naive Bayes model and SVM model accuracy was low. Random Forest model it gives high accuracy separate the successful and unsuccessful crowdfunding campaigns in the given dataset.


  5.3. Reflect on the strengths and weaknesses of each model:

Random Forest:

Strengths:
Often achieves high accuracy.
Handles both categorical and numerical features well.
Can provide feature importance information.

Weaknesses:
Can be computationally expensive for large datasets.
Less interpretable compared to simpler models.

Naive Bayes:

Strengths:
Simple and easy to understand.
Works well with high-dimensional data.
Handles categorical features effectively.

Weaknesses:
Assumes independence between features, which may not always hold true.
Sensitive to irrelevant features.

SVM:

Strengths:
Effective in high-dimensional spaces.
Versatile due to different kernel options.
Can handle non-linearly separable data.

Weaknesses:
Sensitive to the choice of kernel and hyperparameters.
Can be computationally expensive for large datasets.

